### ISSUE TITLE
GA - Pageview Counts - March changes?

### ISSUE THREAD

**Issue by michelle-dooley** [department-of-veterans-affairs/va.gov-team#56710]
## Issue Description
The VA.gov pageview counts in GA have over doubled since the beginning of March.  No changes have been made to GA/GTM configuration in that time period that would attribute to this.  Did some digging to see if it was a new page or event that could be contributing but could not find one thing that stood out.  Reached out to Search Discovery product support and they asked if there had been any changes to the website's code in late February or early March, specifically one that **_writes data to the "document.location" variable in the browser_**?  It looks like whatever happened started around March 7th-9th and has continued.  Need assistance in determining the cause and resolution as this is driving up GA costs.

Some things to consider:

- Any code changes in early March that could of caused this?

- Design System playground impact?

- Preview environments?

Artifacts: 

[GA Pageview report](https://analytics.google.com/analytics/web/#/report/content-pages/a50123418w177519031p184624291/_u.date00=20230101&_u.date01=20230408&_.advseg=&_.useg=&_.sectionId=&explorer-graphOptions.selected=analytics.nthWeek&explorer-graphOptions.compareConcept=analytics.uniquePageviews/) - see page view increase while unique page views remains steady.![image.png](https://images.zenhubusercontent.com/618144c8b17b33313a8c0cbe/326fdce9-4dd0-4986-8b93-c6e6e0545179)

[GA Sessions report](https://analytics.google.com/analytics/web/?authuser=0#/report/content-event-pages/a50123418w177519031p184624291/_u.date00=20230101&_u.date01=20230408&_.sectionId=&_.useg=builtin1&_.gSectionId=site_usage_events&explorer-graphOptions.compareConcept=analytics.avgPageviews&explorer-graphOptions.selected=analytics.nthWeek) - sessions has remained steady but pages/session is increasing.![image.png](https://images.zenhubusercontent.com/618144c8b17b33313a8c0cbe/25d043b4-8942-40e7-95f8-b7457406165d) 

VA.gov monthly summary report for Chris Johnston![image.png](https://images.zenhubusercontent.com/618144c8b17b33313a8c0cbe/82ac381e-29ff-4545-b52f-6468fb0ce3c6)

---
## Tasks
- [ ] _What work is necessary for this story to be completed?_

## Acceptance Criteria
- [ ] _What will be created or happen as a result of this story?_
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by holdenhinkle** [comment]
Some context for page views vs unique page views:

In Google Analytics, "page views" and "unique page views" are two distinct metrics used to measure user engagement with a website. Here's a brief explanation of each:

Page views: A page view is counted each time a user loads or reloads a specific page on a website. It represents the total number of times a page has been viewed, regardless of whether the same user visits the page multiple times. In other words, multiple page views can be recorded for a single user during a single session if they visit the same page more than once.

Unique page views: A unique page view aggregates the number of individual users who have viewed a specific page during a single session. Unlike page views, unique page views only count the first time a user visits a particular page during their session, even if they return to the page multiple times. This metric provides insight into the number of unique users who have viewed a page and helps differentiate between users who view a page once and those who view it multiple times.

In summary, page views represent the total number of times a page has been viewed, while unique page views represent the number of individual users who have viewed a page during a single session. Unique page views provide a better understanding of user engagement, as they take into account the number of distinct users visiting a page rather than just the total number of times a page has been viewed.

**Comment by little-oddball** [comment]
Here is some vets-website commit history around the 3/8 and 3/8 timframe.

https://github.com/department-of-veterans-affairs/vets-website/commits/main?after=1d60527484d6069cd7bf5ebc9647160555752498+314&branch=main&qualified_name=refs%2Fheads%2Fmain

https://github.com/department-of-veterans-affairs/vets-website/commits/main?before=1d60527484d6069cd7bf5ebc9647160555752498+315&branch=main&qualified_name=refs%2Fheads%2Fmain

**Comment by little-oddball** [comment]
Status Rollup:

The team has taken time to walk through platform specific repos and have been unable to find any logic that has been introduced that may be causing the issue being seen. Additionally we have looked through any initiatives that we thought might be having an impact, specifically EKS migration, Preview environments and some DST component updates. These all have comeback negative.

We have additionally investigated the issue from the runtime perspective and did start to see some interesting results related to what appears to be pageview calls being broadcast multiple times. We feel it is narrowed down to these 2 potential items:

- A lot of GA “page_view” events every time a page loads (3-5 each)
- Possible culprit in the user-nav triggering History events which would trigger page_view events

We have also reached out to teams via vfs-all-teams [here](https://dsva.slack.com/archives/CE4304QPK/p1681332006924279) with some very limited response but helpful. In conjunction, we are working with the Identity team related to a place we feel might be the issue and looking for them to perform a revert for us to validate. This is being worked now and will follow up.

**Comment by mchelen-gov** [comment]
Are we able to confirm whether Page Views is the only metric being affected, or if any others (such as Unique Page Views) are impacted as well?

**Comment by michelle-dooley** [comment]
@mchelen-gov - It is just pageviews and we just found the issue and are working to resolve.

**Comment by little-oddball** [comment]
This issue has been resolved with a revert/address of a checkin of the Sign In modal. This component was setup in a manner where multiple calls were being made to GA for the same visit. It is discussed in [this](https://dsva.slack.com/archives/C052BL118S3/p1681399989117739) thread.

[Here](https://github.com/department-of-veterans-affairs/vets-website/pull/23928) is the PR related to.addressing the issue.
